### PR TITLE
[stable/traefik] Allow acme entryPoint to be defined in values

### PR DESCRIPTION
**Pull request by emslade**
Addresses #3437 although I haven't implemented all the options.

### CODE REVIEW

**Comment by rimusz** [comment]
@adeslade would you mind to add the chart name in the PR description, please?

**Comment by emslade** [comment]
Yes of course, sorry.

File: stable/traefik/templates/configmap.yaml [hunk 1]
```diff
@@ -74,6 +74,10 @@ data:
     {{- if .Values.acme.logging }}
     acmeLogging = true
     {{- end }}
+    {{- if eq .Values.acme.httpChallenge.entryPoint "http" }}
```
Comment (emslade) [inline stable/traefik/templates/configmap.yaml]: I'm not sure about this.

Comment (mebrein) [reply stable/traefik/templates/configmap.yaml]: If you specify [acme.httpChallenge], then you also need to specify the name of the entrypoint. Usually most will use 'http' however a name like 'httpxyz' would be perfectly possible too. So if you start using httpChallenge, you could check a second parameter with the name of the entrypoint. Having said that I did not see any examples use a name other than 'http'.

Comment (emslade) [reply stable/traefik/templates/configmap.yaml]: Ah okay, good to know. I'll change it.

File: stable/traefik/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: traefik
-version: 1.17.1
-appVersion: 1.4.6
+version: 1.17.2
+appVersion: 1.5.1
```
Comment (emslade) [inline stable/traefik/Chart.yaml:4]: I needed to update the `traefik` version for the HTTP-01 challenge (https://github.com/containous/traefik/pull/2701). What happens to the chart version in this circumstance?

Comment (mebrein) [reply stable/traefik/Chart.yaml:4]: I configured the current stable to get traefik up-and-running. Then I copied all files locally and I tried your configuration, also change imageTag in values.yaml, from 1.4.6 to 1.5.1. Then I issued a helm upgrade command. I visited the pod logs. First a message 'acme: Could not find solver for: dns-01' followed by a 'Trying to solve HTTP-01' and finally 'Server responded with a certificate.' Apparantly it first tries to use the dns solver. But then it successfully uses the http solver, so at my side a success!

**Comment by krancour** [comment]
This appears to be a duplicate of #3347 

I'd prefer to keep #3347 open for consideration and close this one since #3347 was opened first and is also more closely aligned with a complementary PR (#1299) that adds support for the DNS challenge.

I move to close this PR. @adeslade thank you for the effort nonetheless. Perhaps you'd like to follow #3347 and be part of that conversation?

**Comment by emslade** [comment]
Makes total sense. Should've done a better job of finding existing PRs. Thanks for letting me know.

**Pull request closed without merging**